spec/index.bs

```diff
@@ -1324,7 +1330,110 @@ dictionary GPUBufferBinding {
 };
 </script>
 
-  * {{GPUBufferBinding/size}}: If undefined, specifies the range starting at {{GPUBufferBinding/offset}} and ending at the end of the buffer.
+  * {{GPUBufferBinding/size}}: If undefined, specifies the range starting at
+      {{GPUBufferBinding/offset}} and ending at the end of the buffer.
+
+A {{GPUBindGroup}} object has the following internal slots:
+
+<dl dfn-type=attribute dfn-for="GPUBindGroup">
+    : <dfn>\[[layout]]</dfn> of type {{GPUBindGroupLayout}}.
+    ::
+        The {{GPUBindGroupLayout}} this {{GPUBindGroup}} associates.
+
+    : <dfn>\[[bindings]]</dfn> of type sequence<{{GPUBindGroupBinding}}>.
+    ::
+        The set of {{GPUBindGroupBinding}}s this {{GPUBindGroup}} describes.
+</dl>
+
+### {{GPUDevice/createBindGroup()|GPUDevice.createBindGroup(GPUBindGroupDescriptor)}} ### {#GPUDevice-createBindGroup}
+
+<div algorithm="GPUDevice.createBindGroup(descriptor)">
+
+The <dfn method for="GPUDevice">createBindGroup(|descriptor|)</dfn> method is used to create
+    {{GPUBindGroup}}s.
+
+1. Ensure [=bind group device validation=] is not violated.
+1. Let |bindGroup| be a new valid {{GPUBindGroup}} object.
+1. Ensure {{GPUBindGroupLayout}} in |descriptor|.{{GPUBindGroupDescriptor/layout}} is valid.
+1. Ensure the number of {{GPUBindGroupLayoutDescriptor/bindings}} of
+    |descriptor|.{{GPUBindGroupDescriptor/layout}} exactly equals to the number of
+    |descriptor|.{{GPUBindGroupDescriptor/bindings}}.
+1. For each {{GPUBindGroupBinding}} |bindingDescriptor| in
+    |descriptor|.{{GPUBindGroupDescriptor/bindings}}:
+    1. Ensure |bindingDescriptor|.{{GPUBindGroupBinding/binding}} does not violate
+        [=bind group binding validation=].
+    1. If |bindingDescriptor|.{{GPUBindGroupBinding/resource}} is {{GPUSampler}}:
```

Comment:
[1] I'd prefer to say "If the bind group layout binding.type is `"sampler"`" (but formalized).

This necessitates having the value of the bind group layout binding, which necessitates doing the match-up of the `binding` numbers. E.g.

```
For each bindingDescriptor in descriptor.bindings
    1.  ...
    1.  Ensure there is exactly one {{BindGroupLayoutBinding}} |layoutBinding| in
        |descriptor|.layout.[[bindings]] such that layoutBinding.binding == bindingDescriptor.binding.
```
or something like that. Then you can ref |layoutBinding| later.

Reply:
Ack & Done. Good suggestion!